config/config.go

```diff
@@ -350,5 +351,10 @@ func ReadConfigFile(configPath string, contextSet bool) ([]byte, error) {
 
 // WriteConfigFile writes the config file to disk with preset permission 0644
 func WriteConfigFile(configPath string, data []byte) error {
-	return ioutil.WriteFile(configPath, data, 0644)
+	err := os.MkdirAll(path.Dir(configPath), 0755)
+	if err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(configPath, data, 0755)
```

Comment:
seems a bit odd to mark this config file as executable, I think setting it to `0644` is more correct.

Reply:
Probably